File: loja-ruby/lib/loja/carrinho.rb
require_relative "produto"

module Loja
  class Carrinho
    def initialize
      @produtos = []
    end

    def adicionar(produto)
      @produtos << produto
    end

    def total
      @produtos.sum(&:preco)
    end

    def quantidade
      @produtos.size
    end
  end
end

File: loja-python/src/loja/carrinho.py
from .produto import Produto

class Carrinho:
    def __init__(self):
        self._produtos: list[Produto] = []

    def adicionar(self, produto: Produto) -> None:
        self._produtos.append(produto)

    def total(self) -> float:
        return sum(p.preco for p in self._produtos)

    def quantidade(self) -> int:
        return len(self._produtos)
